src/Compilers/CSharp/Portable/Compilation/InitializerSemanticModel.cs

```diff
@@ -275,5 +275,20 @@ protected override void AnalyzeBoundNodeNullability(BoundNode boundRoot, Binder
         {
             NullableWalker.AnalyzeWithoutRewrite(Compilation, MemberSymbol, boundRoot, binder, diagnostics, createSnapshots);
         }
+
+        protected override bool IsNullableAnalysisEnabled()
+        {
+            switch (MemberSymbol.Kind)
+            {
+                case SymbolKind.Field:
+                case SymbolKind.Property:
+                    return MemberSymbol.ContainingType is SourceMemberContainerTypeSymbol type && type.IsNullableEnabledForConstructorsAndInitializers(useStatic: MemberSymbol.IsStatic);
+                case SymbolKind.Parameter:
+                    return Root is ParameterSyntax { Default: { } equalsValue } && Compilation.IsNullableAnalysisEnabledIn(equalsValue);
```

Comment:
>Compilation.IsNullableAnalysisEnabledIn(equalsValue); [](start = 83, length = 53)

It looks like SourceComplexParameterSymbol is using different node. We should avoid trying to duplicate logic in different places, this is very likely to go out of sync, etc. Consider delegating the work whether its initializer should be analyzed to the parameter symbol and using that helper throughout.

Reply:
Or a helper that returns the syntax node to use.

---
In reply to: [550254940](https://github.com/dotnet/roslyn/pull/49953#discussion_r550254940) [](ancestors = 550254940)